### PR TITLE
Fix #19121: Create dynamics with shortcut or through "Add -> Text" menu

### LINKED ISSUE #19121: Dynamics (playable) shortcuts
### Your idea

At the moment ctrl+E doesn't play as dynamics (only shows). 
Although the custom palette is very effective, in the flow of inserting notes it is much more fluid to have a shortcut than to select and go to the sidebar to apply the dynamics.
At least to have the option.
Alternatively shortcut configuration option.
If the user has shortcut for articulations why not for dynamics?

### Problem to be solved

Expression text does not play dynamics.

### Prior art

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CMakeScore**
Resolves: #19121

Hello. We are working on a project to add functionality to an open source project as a student experiment at the University of Tokyo.
We believe that some of the results of this project are practical, so we have created a pull request. We would appreciate it if you could review this if you have the time.
Thanks in advance.

Part 1: this PR
* The first commit added a shortcut to add a dynamics text box and assigned Ctrl+D.
* In the second commit, we changed the code so that the dynamics you type with Ctrl+Shift+Alphabets will be interpreted and reflected in the playback.

Part 2: See #25254

Part 3: See #25255

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Comment by XiaoMigros** [comment]
Perhaps it's best to open a separate PR for each of these commits?

**Comment by doss-eeic2024-03** [comment]
Thanks, I have split it into 3 parts. See #25254 and #25255

**Comment by avvvvve** [comment]
@CMakeScore It would be great if you could also split the two commits here into two separate PRs.

The first commit is nearly ready to be merged, but a few changes need to be made in the Add > Text menus:

- Rename 'Dynamics' to 'Dynamic' since everything else there is not pluralized
- Move 'Dynamic' between 'Staff text' and 'Expression text'
- Each item in this menu has an underlined letter. Please make 'D' the underlined letter for 'Dynamic' and change the underlined letter of 'Chord symbol' from 'D' to 'H' (these are used for [mnemonic access, which I think is broken now](https://github.com/musescore/MuseScore/issues/18832), but we should still include it for consistency)

image

The second commit is a great feature but still has some bugs that need resolving before merging (I can describe these once there's a separate PR for that one, or I might just drop them here once I've written them up).

Thanks for your work on this so far!

**Comment by CMakeScore** [comment]
@avvvvve Thank you for your review!
We've addressed the points you suggested and separated the second commit into its own PR (#25396).

**Review by avvvvve — CHANGES_REQUESTED** [review]
@CMakeScore Thanks for the menu/copy changes! Just noticed one more thing:

There's a dynamics-related preference that was added in 4.4 that should be respected when adding a dynamic via shortcut. It's in Preferences > Note input > Voice assignment.

If it's set to "All voices on the instrument", the dynamic should have its "Voice assignment" property set to "All" (as it does now). If it's set to "Only the voice they are applied to" and I use Ctrl + D on a note in Voice 2, for instance, it's "Voice assignment" should be set to "Voice 2", but now it still just gets set to "All".

In short, it should follow the logic that currently is in place for adding dynamics via palettes.

https://github.com/user-attachments/assets/98d9f3f7-3815-4c6a-b225-d661216d9ea8

**Comment by avvvvve** [comment]
@CMakeScore Thank you for the quick work!
@mathesoncalum this one is ready for code review now

**Review by avvvvve — APPROVED** [review]

**Review by mathesoncalum — APPROVED** [review]

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
